### ISSUE TITLE
handleOpenURL not firing on iOS cold start 

### ISSUE THREAD

**Issue by dustinblanchard** [EddyVerbruggen/Custom-URL-scheme#93]
I'm on Cordova 3.8.0  and my handleOpenURL function is not firing on a cold start of the app. It works fine if the app is already open, and works great on Android. 

I'm assuming it's related to https://issues.apache.org/jira/browse/CB-7606.

**Comment by arulapj** [comment]
tried to change the version to 3.6.3, still its not working on IOS.

**Comment by xale76** [comment]
have  you found a solution? I have the same problem

**Comment by Exilz** [comment]
Same problem here on Android 5.1 and possibly lower versions too.

**Comment by benjick** [comment]
Same with Android 5.0.1

**Comment by EddyVerbruggen** [comment]
If anyone can reproduce this problem with a vanilla cli generated project then I'd be more than happy to take a look. With something like Meteor in the mix you're on your own.

**Comment by alexszilagyi** [comment]
I'm having the same issue with Ionic framework, tried different tips that I've found on web but none of them are working; my scenario throws this error: `Failed to load webpage with error: Frame load interrupted`.

**Comment by martindrapeau** [comment]
I am using cordova 4.0.0-dev and still have this issue. Any progress?

**Comment by EddyVerbruggen** [comment]
@martindrapeau see my comment of July 9th. That offer still stands.

**Comment by martindrapeau** [comment]
Fair enough - I think we are all using frameworks and wrappers. Cocoon.io in my case. I will see what I can do and get back to you...

**Comment by possan** [comment]
As usual, adding a little delay solves the problem, at least for me...

```
function handleOpenURL(url) {
    setTimeout(function() {
        actuallyHandleOpenURL(url);
    }, 500);
}

function actuallyHandleOpenURL(url) {
...
}
```

**Comment by xale76** [comment]
still not working @possan 
is there any solution?

**Comment by EddyVerbruggen** [comment]
What's your setup like? The more details the better.

**Comment by xale76** [comment]
I don't understand the question, sorry eddy :(

I build with Intel XDK cordova solution. It work for android, and not for iOS when the app is shut. Works when the app in in background. Setup is ipa

**Comment by EddyVerbruggen** [comment]
@xale76 You're describing the symptoms of an issue in Cordova-iOS 3.7.0 - which version are you using?

**Comment by xale76** [comment]
cordova cli?
4.1.2

**Comment by EddyVerbruggen** [comment]
No Cordova-iOS, please check it and move to anything else than 3.7.0.

**Comment by xale76** [comment]
where I can find this information?

**Comment by EddyVerbruggen** [comment]
I don't use XDK and please don't mistake me for Google :)

**Comment by vhuerta** [comment]
Same problem, someone fixed it?

**Comment by vhuerta** [comment]
i got the same problem, someone solved it?

**Comment by EddyVerbruggen** [comment]
@vhuerta thanks for asking twice. What version of cordova-ios are you on? Not cordova-cli.

**Comment by vhuerta** [comment]
Im using the 3.9.1 version

**Comment by EddyVerbruggen** [comment]
Can you share your code with me perhaps?

**Comment by vhuerta** [comment]
sure there is
```javascript
window.handleOpenURL = function handleOpenURL(url) {
    setTimeout(function() {
        if(Meteor.isCordova) {
            var host = window.location.protocol + "//" + window.location.host;
            if(url.indexOf('?external')) {
                Router.go(url.replace('saxmovil://', ''));
            }
        }
    }, 500);
};
```

**Comment by EddyVerbruggen** [comment]
Ah Meteor again. Anything in the adb log or js console?

**Comment by Alexintosh** [comment]
Setting a 500 timeout worked for me.

**Comment by mmBs** [comment]
setTimeout does not work for me too. Does anybody have a different solution?

**Comment by Exilz** [comment]
@mmBs here is my modified meteor_cordova_loader. 
I put this in my-meteor-project/cordova-build-override/www/meteor_cordova_loader.js

https://gist.github.com/Exilz/3d50cf7614b982157faa

Cold start works like a charm on Android/iOS regardless of versions and phone's swiftness since it doesn't depend on timeout.

I guess it's quite risky for Meteor, though...

**Comment by mmBs** [comment]
@Exilz thanks for that. I will try to adapt this approach to my implementation - i use backbone.js

**Comment by JeroenSchonenberg** [comment]
I also _HAD_ problems with handleOpenURL not being called on cold start. It worked fine on the iOS simulator but not on the actual device.

In my case I solved it by placing my handleOpenURL()-implementation in a <script>-element inside the body. Like so:

    <body>
        <script type="text/javascript">
            function handleOpenURL(url) {
                setTimeout(function() {
                    alert(url);
                });
            }
        </script>
        
        
    </body>

On rare occasions it still doesn't work but I haven't been able to reproduce those bugs yet...
This issue can also be merged with #2 imho.

**Comment by JeroenSchonenberg** [comment]
Applying this fix from @easyM0de seems to fix the problem completely:  https://github.com/EddyVerbruggen/Custom-URL-scheme/issues/2#issuecomment-152557017

Could this get implemented? Adding the used URL as a global variable seems like a decent fallback (or even default!). In contrary to the openURL-function you can actually rely on the window being defined.

**Comment by alexspeller** [comment]
Nothing here worked for me, but then I deleted [this line](https://github.com/apache/cordova-ios/blob/99f5a3361080786da8653bdda5fbf1b98c1fede3/CordovaLib/Classes/Private/Plugins/CDVHandleOpenURL/CDVHandleOpenURL.m#L52) and everything started working properly

**Comment by wudongyang** [comment]
Nothing here worked for me too.

I fixed it with call the HandleOpenURLNotification after 0.5s in application:openUrl
It works for me now.

	[self performSelector:@selector(delayHandleOpenUrlNotification:) withObject:url afterDelay:0.5f];

	-(void)delayHandleOpenUrlNotification:(NSURL*)url
	{
    	// all plugins will get the notification, and their handlers will be called
    	[[NSNotificationCenter defaultCenter] postNotification:[NSNotification notificationWithName:CDVPluginHandleOpenURLNotification object:url]];
	}

**Comment by lorensr** [comment]
I imagine this would be fixed if `getLastIntent` was implemented for iOS (it was added in #126 as Android-only).

**Comment by northkode** [comment]
happens to me too.. cold start does not fire.. running ios 3.9.1 and straight up cordova build no frameworks other then my own js framework for loading views.

**Comment by esperia** [comment]
I also happened too.
iOS 9.3 (iPhone4S), Cordova 6.3, Node v6.2.2, Cordova ios 4.2.0

**Comment by srameshr** [comment]
Me on IOS 4. Cold start does not seem to work. 
```index.html```

```
<body>
  <script>
    function handleOpenURL(url) {
        setTimeout(function() {
          window.localStorage.setItem("externalLoad", JSON.stringify(url));
        }, 0);
      }
    </script>
</body>
```

**Comment by tavosansal** [comment]
I am using Ember and have found something interesting. I was not able to trigger the `handleOpenURL` method on a cold boot for the life of me. I created a blank ember app with ember-cordova and nothing. Then I created a blank cordova app and sure enough it works. So now I have discovered that when the app is being served locally (`ember cdv:serve`) the event will never trigger on a cold boot. But if you build the app with (`ember cdv:build`) and then run it, the event does trigger. So I am not entirely sure what the difference is or why this works, but thought I would share. If you are not using ember but still serving your app on your local machine somehow, this might be why you can't trigger that event.
